# Titanic Dataset – Data Preparation & Cleaning

## Data Analytics Internship Program – Week 1

### Author
Noor-ul-Ain

### Tools Used

- Python
- Google Colab
- Pandas
- NumPy

### Dataset

Titanic Dataset (Kaggle)

---

## Objective

To clean and prepare the Titanic dataset by identifying data quality issues, handling missing values, verifying data types, improving readability, and preparing the dataset for future analysis.

# Import Required Libraries

In [1]:
import pandas as pd
import numpy as np

# Load Dataset

In [2]:
from google.colab import files

uploaded = files.upload()

Saving Titanic-Dataset.csv to Titanic-Dataset.csv


# Read Dataset

In [3]:
df = pd.read_csv("Titanic-Dataset.csv")

# Dataset Exploration

In [4]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [5]:
df.shape

(891, 12)

In [6]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [8]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [9]:
df.dtypes

,0
PassengerId,int64
Survived,int64
Pclass,int64
Name,object
Sex,object
Age,float64
SibSp,int64
Parch,int64
Ticket,object
Fare,float64


# Data Quality Assessment

In [10]:
df.isnull().sum()

,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,177
SibSp,0
Parch,0
Ticket,0
Fare,0


In [11]:
df.duplicated().sum()


np.int64(0)

In [12]:
df["Sex"].unique()

array(['male', 'female'], dtype=object)

In [13]:
df["Embarked"].unique()

array(['S', 'C', 'Q', nan], dtype=object)

In [14]:
missing_values = df.isnull().sum()

missing_percentage = (missing_values / len(df)) * 100

missing_report = pd.DataFrame({
    "Missing Values": missing_values,
    "Percentage": missing_percentage
})

missing_report

,Missing Values,Percentage
PassengerId,0,0.000000
Survived,0,0.000000
Pclass,0,0.000000
Name,0,0.000000
Sex,0,0.000000
Age,177,19.865320
SibSp,0,0.000000
Parch,0,0.000000
Ticket,0,0.000000
Fare,0,0.000000


# Handling Missing Values

In [15]:
df["Age"] = df["Age"].fillna(df["Age"].median())

In [16]:
df.drop(columns=["Cabin"], inplace=True)

In [17]:
df["Embarked"] = df["Embarked"].fillna(df["Embarked"].mode()[0])

In [18]:
df.isnull().sum()

,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,0
SibSp,0
Parch,0
Ticket,0
Fare,0


# Data Type Verification

In [19]:
df.dtypes

,0
PassengerId,int64
Survived,int64
Pclass,int64
Name,object
Sex,object
Age,float64
SibSp,int64
Parch,int64
Ticket,object
Fare,float64


# Column Renaming

In [20]:
df.rename(columns={
    "PassengerId": "Passenger_ID",
    "Pclass": "Passenger_Class",
    "SibSp": "Siblings_Spouses",
    "Parch": "Parents_Children"
}, inplace=True)

# Final Dataset Verification

In [21]:
df.columns

Index(['Passenger_ID', 'Survived', 'Passenger_Class', 'Name', 'Sex', 'Age',
       'Siblings_Spouses', 'Parents_Children', 'Ticket', 'Fare', 'Embarked'],
      dtype='object')

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Passenger_ID      891 non-null    int64  
 1   Survived          891 non-null    int64  
 2   Passenger_Class   891 non-null    int64  
 3   Name              891 non-null    object 
 4   Sex               891 non-null    object 
 5   Age               891 non-null    float64
 6   Siblings_Spouses  891 non-null    int64  
 7   Parents_Children  891 non-null    int64  
 8   Ticket            891 non-null    object 
 9   Fare              891 non-null    float64
 10  Embarked          891 non-null    object 
dtypes: float64(2), int64(5), object(4)
memory usage: 76.7+ KB


In [23]:
df.isnull().sum()

,0
Passenger_ID,0
Survived,0
Passenger_Class,0
Name,0
Sex,0
Age,0
Siblings_Spouses,0
Parents_Children,0
Ticket,0
Fare,0


In [24]:
df.shape

(891, 11)

# Export Clean Dataset

In [25]:
df.to_csv("cleaned_titanic.csv", index=False)

In [26]:
from google.colab import files

files.download("cleaned_titanic.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

## Project Summary

- Successfully imported and explored the Titanic dataset using Python and Pandas.
- Identified missing values in the Age, Cabin, and Embarked columns during data quality assessment.
- Replaced missing values in the Age column using the median to minimize the impact of outliers.
- Filled missing values in the Embarked column using the mode, as it is a categorical feature.
- Removed the Cabin column because it contained more than 77% missing values, making it unsuitable for reliable analysis.
- Checked for duplicate records and confirmed that the dataset contained no duplicates.
- Verified that all data types were appropriate for analysis, so no type conversions were required.
- Renamed selected columns to improve readability and maintain consistency.
- Verified that the cleaned dataset contained no missing values and was ready for future analysis.
- Exported the final cleaned dataset as cleaned_titanic.csv for further analytical tasks.